src/dns.rs

```diff
@@ -11,7 +11,7 @@ pub trait ToIpAddr {
 }
 
 pub trait ToSocketAddr {
-    fn to_socket_addr(&self) -> SocketAddr;
+    fn to_socket_addr(&self, dns: &Dns) -> SocketAddr;
```

Comment:
Is it concerning that this doesn't line up with tokio's signature (where DNS is global)?

Reply:
A bit, but the trait is opaque so I'm good to follow up on this.